block-producer/src/lib.rs

```diff
@@ -8,13 +8,15 @@ use tokio::sync::RwLock;
 pub mod test_utils;
 
 pub mod batch_builder;
+pub mod block;
 pub mod block_builder;
+pub mod cli;
+pub mod config;
```

Comment:
Ideally these would not go in the library crate (to keep "core logic" separate from "cli logic"), but I'm not sure the best way to fix it. Our options are:

+ Create a new crate for the cli
+ Put the cli under `src/bin/<executable-name>`
+ Something else?

Reply:
We can probably do something similar to what we do in [Miden VM](https://github.com/0xPolygonMiden/miden-vm/blob/main/miden/src/main.rs) (i.e., import `cli` module directly in `main.rs`). If we go this route, we should do the same for other components (store, RPC), and I'd do this in a separate PR. So, let's create an issue for this.